File: algorithm-challenges/vowelcount/vowelcount-solved.js
// Write a function that takes a string input: 'str'
// Return the number / count of vowels in the input string.
// For the purpose of this assignment, consider 'a', 'e', 'i', 'o' and 'u'
// as vowels
// the input string will consist of lower case letters

function countTheVowels(str) {
    // initialize a counter so that we can count the vowels
    var vowelsCount = 0;

    // set up an array with vowels that we can loop through and check against
    var vowels = ['a', 'e', 'i', 'o', 'u'];

    // loop through the input string
    for (var i = 0; i < str.length; i++) {
        // for each item in the array (i), loop through the vowels array and see if it matches
        for (var j = 0; j < vowels.length; j++) {
            // if there is a match, increase the counter by 1
            if (str[i] === vowels[j]) {
                vowelsCount++;
            }
        }
    }
    // return the vowel counter
    return vowelsCount;
}

console.log(countTheVowels('this is a string'));

File: data-visualization/whiteboarding/e-2-vowelcount/vowelcount-solved.py
#  Write a function that takes a string input: 'str'
#  Return the number / count of vowels in the input string.
#  For the purpose of this assignment, consider 'a', 'e', 'i', 'o' and 'u'
#  as vowels
#  the input string will consist of lower case letters

def count_the_vowels(str): 

    # initialize a counter variable  to 0 
    num_vowels = 0

    # loop through each character in the string
    for char in str:

        #  if the char matches one of the  vowels in upper case of lowercase
        if char in "aeiouAEIOU":

            # increment counter by one
            num_vowels = num_vowels + 1

    # return  the  counter
    return num_vowels


print(count_the_vowels('this is a string'))